cdc/sink/dmlsink/mq/worker.go

```diff
@@ -291,12 +301,30 @@ func (w *worker) sendMessages(ctx context.Context) error {
 					zap.String("changefeed", w.changeFeedID.ID))
 				return nil
 			}
-			if err := future.Ready(ctx); err != nil {
+			if err = future.Ready(ctx); err != nil {
 				return errors.Trace(err)
 			}
 			for _, message := range future.Messages {
+				if message.ClaimCheckFileName != "" {
+					// send the message to the external storage.
+					if err = w.claimCheck.WriteMessage(ctx, message.Key, message.Value, message.ClaimCheckFileName); err != nil {
+						log.Error("send message to the external claim check storage failed",
+							zap.String("namespace", w.changeFeedID.Namespace),
+							zap.String("changefeed", w.changeFeedID.ID),
+							zap.String("filename", message.ClaimCheckFileName),
+							zap.Error(err))
+						return errors.Trace(err)
+					}
+					// create the location message which contain the external storage location of the message.
+					locationMessage, err := w.claimCheckEncoder.NewClaimCheckMessage(message.Event, message.Callback)
+					if err != nil {
+						return errors.Trace(err)
+					}
+					message = locationMessage
+				}
+				// normal message, just send it to the kafka.
 				start := time.Now()
-				if err := w.statistics.RecordBatchExecution(func() (int, error) {
+				if err = w.statistics.RecordBatchExecution(func() (int, error) {
 					if err := w.producer.AsyncSendMessage(ctx, future.Topic, future.Partition, message); err != nil {
```

Comment:
Does the message sent to Kafka  only contain the url location without pk ?

Reply:
It contain both url and handle key columns.